pandas/core/reshape/concat.py

```diff
@@ -362,17 +364,22 @@ def __init__(
         self.objs = objs
 
         # Standardize axis parameter to int
-        if isinstance(sample, Series):
+        # TODO: Should this really require a class import?
+        """
+        if isinstance(sample, ABCSeries):
             axis = DataFrame._get_axis_number(axis)
         else:
             axis = sample._get_axis_number(axis)
+        """
+        if not isinstance(axis, int):
+            axis = {"index": 0, "columns": 1}[axis]
```

Comment:
looks like you need the alias "rows" here

Reply:
Ha yep - nice little surprise there. Actually taking a look at the axis handling atm; might clean up and do as a pre-cursor.

Reply:
Going to sit on this PR for a bit. Somewhat of a rabbit hole but I think will make sense to get traction on #29140 and simplify the index handling instead of doing this as a one-off, especially since there's not much of a rush here

Reply:
Is fixing this more complicated than just adding the "rows" alias to that dict?  runtime imports of concat are a non-trivial pain point for me when trying to reason about code, so i'd like to see this go through

Reply:
No that's it. I was hoping to do a more general utility function for axis validation (see #29140) but I can just add this here as well if you find it helpful. Not the worst cruft...